### DISCUSSION TITLE
How to visit any node in the validator?

### DISCUSSION THREAD

**Discussion by trusktr** [eclipse-langium/langium#982] [Q&A]
For example something like

```ts
	const checks: ValidationChecks<MyLangAstType> = {
		<what-goes-here>: checkAnyNode,
	}
```

Similarly is there separate builtin API to visit all AST nodes? Or do we implement a custom traverser ourselves?

Just trying to get ideas based on what is possible.

**Comment by msujew** [comment]
I've created https://github.com/langium/langium/issues/983 to track this. I believe this would make sense as a general validator feature, but as of now, you would need to override the [`DefaultDocumentValidator#validateAst`](https://github.com/langium/langium/blob/712bd01d58dc3c683ccbb27f813e1c0e7fc941f6/packages/langium/src/validation/document-validator.ts#L138-L152) method.
